src/nvim/lua/vim.lua

```diff
@@ -55,29 +55,30 @@ end
 
 function vim._load_package(name)
   local basename = name:gsub('%.', '/')
-  local paths = {"lua/"..basename..".lua", "lua/"..basename.."/init.lua"}
-  for _,path in ipairs(paths) do
-    local found = vim.api.nvim_get_runtime_file(path, false)
-    if #found > 0 then
-      local f, err = loadfile(found[1])
-      return f or error(err)
-    end
+  local paths = "lua/"..basename..".lua lua/"..basename.."/init.lua"
+  local found = vim.api.nvim_get_runtime_file(paths, false)
+  if #found > 0 then
+    local f, err = loadfile(found[1])
+    return f or error(err)
   end
 
+  local paths = {}
   for _,trail in ipairs(vim._so_trails) do
-    local path = "lua"..trail:gsub('?', basename) -- so_trails contains a leading slash
-    local found = vim.api.nvim_get_runtime_file(path, false)
-    if #found > 0 then
-      -- Making function name in Lua 5.1 (see src/loadlib.c:mkfuncname) is
-      -- a) strip prefix up to and including the first dash, if any
-      -- b) replace all dots by underscores
-      -- c) prepend "luaopen_"
-      -- So "foo-bar.baz" should result in "luaopen_bar_baz"
-      local dash = name:find("-", 1, true)
-      local modname = dash and name:sub(dash + 1) or name
-      local f, err = package.loadlib(found[1], "luaopen_"..modname:gsub("%.", "_"))
-      return f or error(err)
-    end
+    table.insert(paths, "lua"..trail)
+  end
+  local paths = table.concat(paths, " ")
```

Comment:
`local` is like redefining the variable, are we doing this commonly? I would expect lualint to flag this.
```suggestion
  paths = table.concat(paths, " ")
```

Reply:
I wasn't sure about that when I wrote it. I wrote it that way because Lua is a bit too happy to bind to a global if you're not careful and it seemed reasonable from that angle. Does Lua (or LuaJIT) do something undesirable due to this, or is it purely a matter of avoiding redundancy for style purposes?

As the linter complains, I'll remove the redundant `local`.

Reply:
> Lua is a bit too happy to bind to a global if you're not careful and it seemed reasonable from that angle.

linter should catch that

Reply:
I was going to ask about that, but it looks like I made some false assumptions about the capabilities of the linter.